packages/application-config/src/process-config.ts

```diff
@@ -53,16 +63,19 @@ const processConfig = ({
   applicationPath = fs.realpathSync(process.cwd()),
 }: ProcessConfigOptions = {}): ApplicationRuntimeConfig => {
   if (cachedConfig && !disableCache) return cachedConfig;
+  const { filepath, config: rawConfig } = loadConfig(applicationPath);
+  const configType = getConfigurationType(
+    parseFilePath(filepath).name
+  );
 
-  const rawConfig = loadConfig(applicationPath);
-  validateConfig(rawConfig);
+  validateConfig(configType, rawConfig);
   const appConfig =
     substituteVariablePlaceholders<JSONSchemaForCustomApplicationConfigurationFiles>(
       rawConfig,
       { applicationPath, processEnv }
     );
-  const customApplicationData =
-    transformCustomApplicationConfigToData(appConfig);
+
+  const configurationData = transformConfigurationToData(configType, appConfig);
```

Comment:
We provide the config type to the transform function so it know which output we expect (Custom Apps or Custom Views)